src/util/figures.py

```diff
@@ -115,21 +116,33 @@ def bar_total_x_by_ybins(df: pd.DataFrame, total_col: str, group_by_cols: list[s
         group_names = [meta.get_friendly_name(col) for col in group_by_cols]
         fig_params["title"] = f"Total {meta.get_friendly_name(total_col)} by {', '.join(group_names)} Bins"
 
+    # minimal change: allow color override; if fcode_desc is used, apply DEFAULT_FCODE_COLOR_MAP
```

Comment:
I think we should pass the color ramp into the function, not detect what type of graph it is inside a generic graphing function. That way we can just pass the appropriate color ramp instead of having logic to detect if this is an fcode graph.